File: examples/08_animation/asset/script/main.lua
local texture = sn.Texture.new()
texture:fill(sn.Color.new(1, 1, 1, 1))
local model = sn.Model.new()
model:load("BrainStem.glb")
local draw3d = sn.Draw3D.new(texture)
draw3d.position = sn.Vec3.new(0, 0, 0)
draw3d.model = model
draw3d.material = model:getMaterial()

local vertex_shader = sn.Shader.new()
vertex_shader:compileLoadVertexShader("skinning.slang")
local fragment_shader = sn.Shader.new()
fragment_shader:compileLoadFragmentShader("skinning.slang")

local pipeline = sn.GraphicsPipeline.new()
pipeline:setVertexShader(vertex_shader)
pipeline:setFragmentShader(fragment_shader)
pipeline:setEnableDepthTest(true)
pipeline:setEnableAnimation(true)
pipeline:build()
sn.Graphics.bindPipeline(pipeline)


sn.Graphics.getCamera():lookat(sn.Vec3.new(0, -3, 1), sn.Vec3.new(0, 0, 1), sn.Vec3.new(0, 0, 1))

model:play(0)
function update()
    if sn.Keyboard.isPressed(sn.Keyboard.ESCAPE) then
        sn.Script.load("main", ".")
    end
    model:update(sn.Time.delta())
end

function draw()
    sn.Graphics.setUniformData(1, model:getBoneUniformData())
    -- Draw texture
    sn.Graphics.drawModel(model, sn.Transform.new(), draw3d.material)
end

File: examples/08_animation/asset/script/main.py
import sinen as sn

texture = sn.Texture()
texture.fill(sn.Color(1, 1, 1, 1))
model = sn.Model()
model.load("BrainStem.glb")

vertex_shader = sn.Shader()
vertex_shader.compile_and_load_vertex_shader("skinning.slang")
fragment_shader = sn.Shader()
fragment_shader.compile_and_load_fragment_shader("skinning.slang")

pipeline = sn.GraphicsPipeline()
pipeline.set_vertex_shader(vertex_shader)
pipeline.set_fragment_shader(fragment_shader)
pipeline.set_enable_animation(True)
pipeline.set_enable_depth_test(True)
pipeline.build()
sn.Graphics.bind_pipeline(pipeline)


sn.Graphics.get_camera().lookat(sn.Vec3(0, -3, 1), sn.Vec3(0, 0, 1), sn.Vec3(0, 0, 1))

model.play(0)


transform = sn.Transform()
material = sn.Material()
material.append_texture(texture)


def update():
    if sn.Keyboard.is_pressed(sn.Keyboard.ESCAPE):
        sn.Script.load("main", ".")
    model.update(sn.Time.delta())


def draw():
    sn.Graphics.set_uniform_data(1, model.get_bone_uniform_data())
    # Draw texture
    sn.Graphics.draw_model(model, transform, material)
